Update nft-rfc-008.md (#12)
interNFT/nft-rfc: nft-rfc-008.md

   If the criteria, as definited in the schema template, are not met to the satisfaction of the NFT issuing authority 
   they can revoke the minting capability of the Claim Issuer and/or burn any minted NFT(s).
   Some example criteria that could be specified:
   * Claim Issuer must provide a certification from a certifying agency recognised by the NFT issuing authority
   * Claim Issuer must have their Claim validated by a recognized evaluating agency
   * Claim Issuer must provide evidence as referrenced by the Claim Evaluation
   * Claim must be submitted within a certain timeframe of the claim issuing date
   * etc
2. Once the NFT issuing authoriy has specified their schema a Claim Issuer can now evaluate the published schema and 
   begin the process of collecting any credentials they may need to satisfy the criteria set forth in the schema. For
   the Claim Issuer this may include:
   * Contacting a recognized certification agency who can inspect their infrastructure and provide an inspection certificate.
   * Finding a certified Claim Verification Agent to evaluate their claims
   * Ensure the can provide the evidence needed by the Claim Verification Agent
   * etc

> 3. Now that the Claim Issuer has collected the necissary pieces to fullfil the NFT issuing authority's criteria they can now

Reviewer: Correct spelling of `necessary`

> 3. Now that the Claim Issuer has collected the necissary pieces to fullfil the NFT issuing authority's criteria they can now

Reviewer 2: ```suggestion
3. Now that the Claim Issuer has collected the necessary pieces to fulfill the NFT issuing authority's criteria they can now
```

>    provide an initial set of proof of criteria fulfillment to the NFT issuing authority, void of any claim, and receive

Reviewer: Rather than 'void of any claim', we should introduce the `caveats` terminology from the Z-Cap spec

>    provide an initial set of proof of criteria fulfillment to the NFT issuing authority, void of any claim, and receive

Reviewer 2: ```suggestion
   provide an initial set of proof of criteria fulfillment to the NFT issuing authority and receive
```

   a capability to mint the NFT specified by the schema from the NFT issuing authority. This capability allows the Claim
   Issuer to mint an NFT assuming they provide verifiable proof that the schema criteria has been met. 
4. After the Claim Issuer receives their minting capability they may now begin issuing the claims they will be making. Note

>    that this process could and likely will be asynchronos in the a Claim Issuer can start issuing claims prior to any 

Reviewer: Sp `asynchronous`

>    that this process could and likely will be asynchronos in the a Claim Issuer can start issuing claims prior to any 

Reviewer 2: ```suggestion
   that this process could and likely will be asynchronous in the a Claim Issuer can start issuing claims prior to any 
```

   certification process taking place, as long as this does not violate any of the NFT schema's criteria. 
   Some claim examples:
   * Production 1000MWh of solar energy
   * Increased performance of a student by a letter grade over a semester
   * Removal of 50tons of carbon via carbon capture
5. Once the claims have been issued, the Claim Issuer now begins the process of turning their claims into an NFT. This begins

>    by first going through the necissary steps to satisfy the NFT schema criteria and may include:

Reviewer: Sp `necessary`

>    by first going through the necissary steps to satisfy the NFT schema criteria and may include:

Reviewer 2: ```suggestion
   by first going through the necessary steps to satisfy the NFT schema criteria and may include:
```

   * A reference to a certification made by a recognized agency
   * A reference to a cerfied Claim Verification Agent's evaluation of the claim
   * etc

> 6. Only after satisfying criteria as denoted by the NFT schema criteria can the Claim Issuer now take the set

Reviewer: Include this must meet the `caveats` and `capabilityInvocation` requirements

> 6. Only after satisfying criteria as denoted by the NFT schema criteria can the Claim Issuer now take the set

Reviewer 2: ```suggestion
6. Only after satisfying criteria as denoted by the NFT schema criteria, and any caveats attached 
to the capability, can the Claim Issuer now take the set
```

   of data--this being the claim itself and the set of data which satisfies the schema criteria--and use this to mint an NFT.

> 7. This NFT can now be governed by the particular NFT requirements (e.g. mintable, tradable, non-fungable, etc) with the NFT

Reviewer: Change 'non-fungible' to 'immuatable'

> 7. This NFT can now be governed by the particular NFT requirements (e.g. mintable, tradable, non-fungable, etc) with the NFT

Reviewer 2: ```suggestion
7. This NFT can now be governed by the particular NFT requirements (e.g. mintable, tradable, immutable, etc) with the NFT
```

   attestations made by the Claim Issuer against the specified NFT schema criteria.    

## Carbon Credits Example
To walk through the above Verification Process we are going to use an example of a company, HydroElec Inc., who has recently upgraded 
a dam to support the production of hydroelectric power and is going through the process of claiming 100 tons of Carbon Credits from 
the UN’s International Climate Change Authority (UNFCCC).

Throughout this overview there will be reference to four different Verifiable Credentials, which will be denoted the first time they are 
used as VC1, VC2, VC3, VC4. The set of these five VCs will be utilized in the minting of a NFT with its associated metadata.

In this example we will use the name of a few companies representing the different authorities involved in the process: 
* __HydroElec__ Inc is the clean energy producer who issues a renewable energy certificate (REC) claim‒in the format of VC2, of having 
  produced a number of Mw Hours of clean energy through its certified hydroelectric project. 
* __CleanEnergy Certifier__ is the certifying authority who certifies HydroElec Inc’s new hydroelectric dam project and issues VC1

> * __CarbonAudit__ is the REC Claim Verification Agent who evaluates the producer’s CER claim and uses a capability issued by the UNFCCC

Reviewer: 'REC claim' not 'CER Claim'

> * __CarbonAudit__ is the REC Claim Verification Agent who evaluates the producer’s CER claim and uses a capability issued by the UNFCCC

Reviewer 2: ```suggestion
* __CarbonAudit__ is the REC Claim Verification Agent who evaluates the producer’s REC claim and uses a capability issued by the UNFCCC
```

      - Claim Verification Agent must include a hashed reference to each Claim they processed
      - Claim Verification Agent must include a timestamp in ISO 8601 format for when each Claim was processed
   * Energy Claims must be made in units of MWh
   * Energy Claims must include the time range the energy was produced in and must follow ISO 8601
   * Claim Issuer must include a specific IOT device reference the claim was generated from
   * Claim Issuer must include last service date in ISO 8601 format for each IoT device
2. CleanEnergy Certifier gets certified by the UNFCCC to act as a project auditor for the purposes of certifying a project to be able to make
   REC claims and receives VC4 from the UNFCCC. At the same time CarbonAudit gets certified by the UNFCCC as a Claim Verification Agent for 
   these new REC NFTs and receives a capability from UNFCCC to issue an Evaluation Report Credentials (VC3). 
3. HydroElec get CleanEnergy Certifier to audit the new electric dam they just converted and receives a Certifying Credential (VC1) from 
   CleanEnergy Certifier which includes, by reference, CleanEnergy Certifier's Certification Credential (VC4) they received from UNFCCC.

> 4. HydroElec submits their Certifying Credential (VC1) to the UNFCCC and receives the capability of minting REC NFTs. 

Reviewer: This step might be redundant and can be further decentralised, by adding a `caveat` to the authorisation provided by UNFCC to the producer, that they must present proof of VC1 to be able to `evokeCapability` `mintNft`

5. HydroElec begins producing renwable energy claims and stores them, by reference, on the Clean Energy Blockchain. Enlisting CarbonAudit, they
   provide a capability to CarbonAudit to access the referenced renwable energy claims. 
6. CarbonAudit processes the claims as they are registered to the Clean Energy Blockchain and, as they only issue Evaluation Report credentials (VC3s).
7. Once CarbonAudit has evaluated 1,000MWh worth of claims, they bundle their evaluations in a VC3, which contains: 

>    * Reference to each claim evaluated

Reviewer 2: ```suggestion
   * Reference to VC1 describing the project
   * Reference to each claim evaluated
```

>    * Outcome and reasoning for each claim evaluated

Reviewer: VC3 should include a reference to VC1 in order to meet the evocation caveat

   these new REC NFTs and receives a capability from UNFCCC to issue an Evaluation Report Credentials (VC3). 
3. HydroElec get CleanEnergy Certifier to audit the new electric dam they just converted and receives a Certifying Credential (VC1) from 
   CleanEnergy Certifier which includes, by reference, CleanEnergy Certifier's Certification Credential (VC4) they received from UNFCCC.
4. HydroElec submits their Certifying Credential (VC1) to the UNFCCC and receives the capability of minting REC NFTs. 
5. HydroElec begins producing renwable energy claims and stores them, by reference, on the Clean Energy Blockchain. Enlisting CarbonAudit, they
   provide a capability to CarbonAudit to access the referenced renwable energy claims. 
6. CarbonAudit processes the claims as they are registered to the Clean Energy Blockchain and, as they only issue Evaluation Report credentials (VC3s).
7. Once CarbonAudit has evaluated 1,000MWh worth of claims, they bundle their evaluations in a VC3, which contains: 
   * Reference to VC1 describing the project
   * Reference to each claim evaluated

>    * Outcome and reasoning for each claim evaluated

Reviewer 2: ```suggestion
   * Outcome and reasoning for each claim evaluated WHOOHOOO
```
